### PR TITLE
#651: FEAT: Added support for HMSET 

### LINKED ISSUE #651: Add support for command `HMSET`
Add support for the `HMSET` command in DiceDB similar to the [`HMSET` command in Redis](https://redis.io/docs/latest/commands/command-list/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

### PR DESCRIPTION
**Pull request by raghavbabbar**
Closes #651

### CODE REVIEW

**Comment by raghavbabbar** [comment]
@JyotinderSingh  please review this. lint is failing because of existing issue.

File: integration_tests/commands/async/hmset_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package async
+
+import (
+	"gotest.tools/v3/assert"
+	"testing"
+)
+
+func TestHMSET(t *testing.T) {
```
Comment (JyotinderSingh) [inline integration_tests/commands/async/hmset_test.go:8]: Please add a test case where you are setting multiple fields in a single `HMSET` command, then verify that all of those fields contain the new value. To make this test even more robust, some of these fields could be pre-existing, while others can be new.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]

File: integration_tests/commands/async/hmset_test.go (hunk 1 above)
Comment (raghavbabbar) [reply integration_tests/commands/async/hmset_test.go:8]: done, sorry for missing this.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for `HMSET`, @raghavbabbar! LGTM.

**Pull request merged**